crates/lib/src/install.rs

```diff
@@ -2318,29 +2319,34 @@ pub(crate) async fn install_reset(opts: InstallResetOpts) -> Result<()> {
 
     // Compute the kernel arguments to inherit. By default, that's only those involved
     // in the root filesystem.
-    let root_kargs = if opts.no_root_kargs {
-        Vec::new()
-    } else {
+    let mut kargs = crate::bootc_kargs::get_kargs_in_root(rootfs, std::env::consts::ARCH)?;
+
+    // Extend with root kargs
+    if !opts.no_root_kargs {
         let bootcfg = booted_ostree
             .deployment
             .bootconfig()
             .ok_or_else(|| anyhow!("Missing bootcfg for booted deployment"))?;
         if let Some(options) = bootcfg.get("options") {
-            let options = options.split_ascii_whitespace().collect::<Vec<_>>();
-            crate::bootc_kargs::root_args_from_cmdline(&options)
-                .into_iter()
-                .map(ToOwned::to_owned)
-                .collect::<Vec<_>>()
-        } else {
-            Vec::new()
+            let options_cmdline = Cmdline::from(options.as_str());
+            let root_kargs = crate::bootc_kargs::root_args_from_cmdline(&options_cmdline);
+            kargs.extend(&root_kargs);
         }
-    };
+    }
 
-    let kargs = crate::bootc_kargs::get_kargs_in_root(rootfs, std::env::consts::ARCH)?
-        .into_iter()
-        .chain(root_kargs.into_iter())
-        .chain(opts.karg.unwrap_or_default())
-        .collect::<Vec<_>>();
+    // Extend with user-provided kargs
+    if let Some(user_kargs) = opts.karg {
+        if !user_kargs.is_empty() {
```

Comment:
I don't think we need this condition anymore since we're no longer incorrectly mapping an empty vector to an empty string.

Reply:
:+1: Removed the `is_empty()` check.